### PR TITLE
Add basic cmd and pkg for regitstration controller/agent

### PR DESCRIPTION
**Pull request by qiujian16**
This is to add leader election, healthz, metrics and event

### CODE REVIEW

File: cmd/registration/main.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import (
+	goflag "flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
```
Comment (deads2k) [inline cmd/registration/main.go:12]: @pmorie you think the gaps here still have value?  In the past go imports didn't handle them that well.

Comment (pmorie) [reply cmd/registration/main.go:12]: I like the gaps, but i've found in large projects they're so inconsistent that they don't necessarily add a lot of value as a convention

**Comment by pmorie** [comment]
This built for me locally, I'll merge and add the build machinery on top

**Pull request merged**